umap/static/umap/js/modules/sync/updaters.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * This file contains the updaters: classes that are able to convert messages
+ * received from another party (or the server) to changes on the map.
+ */
+
+class BaseUpdater {
+  constructor(map) {
+    this.map = map
+  }
+
+  updateObjectValue(obj, key, value) {
+    const parts = key.split('.')
+    const lastKey = parts.pop()
+
+    // Reduce the current list of attributes,
+    // to find the object to set the property onto
+    const objectToSet = parts.reduce((currentObj, part) => {
+      if (part in currentObj) return currentObj[part]
+    }, obj)
+
+    // In case the given path doesn't exist, stop here
+    if (objectToSet === undefined) return
+
+    // Set the value (or delete it)
+    if (typeof value === 'undefined') {
+      delete objectToSet[lastKey]
+    } else {
+      objectToSet[lastKey] = value
+    }
+  }
+
+  getDataLayerFromID(layerId) {
+    if (layerId) return this.map.getDataLayerByUmapId(layerId)
+    return this.map.defaultEditDataLayer()
```

Comment:
It's currerntly used when there are no layers, but the edits still need to be synced on the different connected peers.

Reply:
I don't get it, let's have a talk next week :)

Reply:
Let me try to explain this better when I believe this is useful:

When creating a new map, it is possible to start making changes on a layer that doesn't already exist on the server (when no layer has been saved yet). In this case, we still want the changes to be synced between the peers.

Because the IDs are decided by the server and not by the client, it is not possible to send this information alongside the rest of the data. In the meantime, the "default" datalayer is selected.

Does it makes more sense?